README.md

```diff
@@ -73,26 +104,150 @@ The system uses dedicated queues for managing different job phases:
 
 ## 🛠️ Technical Requirements
 
-### Prerequisites
+### System Dependencies
+
+> For macOS users, use `brew install` instead of `apt install` for these dependencies.
+
+- Build essentials (`build-essential`)
+- OpenSSL (`libssl-dev`)
+- Package config (`pkg-config`)
+- Python 3.9 with development files
+- GMP library (`libgmp-dev`)
+
+### Core Dependencies
+
+- [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [Rust](https://www.rust-lang.org/tools/install)
+- [Madara Node](https://github.com/madara-alliance/madara)
+  - Required for block processing
+  - Follow setup instructions at [Madara Documentation](https://github.com/madara-alliance/madara)
+- Prover Service (ATLANTIC)
+- MongoDB for job management
+- AWS services (or Localstack for local development):
+  - SQS for queues
+  - S3 for data storage
+  - SNS for alerts
+  - EventBridge for scheduling
+
+> 🚨 **Important Note**: Currently, Madara doesn't support the `get_storage_proof` endpoint.
+>
+> Therefore, you need to run Pathfinder alongside Madara:
+>
+> - Madara will run in sequencer mode
+> - Pathfinder will sync with Madara
+> - The orchestrator will use Pathfinder's RPC URL for SNOS and state update fetching
+>
+> This setup ensures proper functionality until the `get_storage_proof` endpoint is implemented in Madara.
```

Comment:
to be precise, till SNOS is adapted to work with `get_storage_proof` RPC endpoint

Reply:
done